library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java

```diff
@@ -66,9 +68,35 @@ public static DownloadManagerBuilder newInstance(Context context, Handler callba
         FileSizeRequester fileSizeRequester = new NetworkFileSizeRequester(httpClient, requestCreator);
         FileDownloader fileDownloader = new NetworkFileDownloader(httpClient, requestCreator);
 
-        NotificationChannelCreator notificationChannelCreator = new DownloadNotificationChannelCreator(context.getResources());
+        NotificationCustomiser<DownloadBatchStatus> notificationCustomiser = new NotificationCustomiser<DownloadBatchStatus>() {
 
-        NotificationCreator<DownloadBatchStatus> downloadBatchNotification = new DownloadBatchNotification(context, notificationIcon);
+            private static final boolean NOT_INDETERMINATE = false;
+
+            @Override
+            public Notification customNotificationFrom(NotificationCompat.Builder builder, DownloadBatchStatus payload) {
+                DownloadBatchTitle downloadBatchTitle = payload.getDownloadBatchTitle();
```

Comment:
🏈

Happy to move this wherever we think is best.

Reply:
I think I would prefer these as anonymous inner classes. I think we are at risk of making these constructors super hard to follow. If we pull them out into anonymous classes at least then they are self contained. What do you think?